ee/tuf/finalize_linux.go

```diff
@@ -0,0 +1,82 @@
+//go:build linux
+// +build linux
+
+package tuf
+
+import (
+	"bytes"
+	"context"
+	"debug/elf"
+	"errors"
+	"fmt"
+	"path/filepath"
+	"time"
+
+	"github.com/kolide/launcher/pkg/allowedcmd"
+)
+
+// On NixOS, we have to set the interpreter for any non-NixOS executable we want to
+// run. This means the binaries that our updater downloads.
+// See: https://unix.stackexchange.com/a/522823
+func patchExecutable(executableLocation string) error {
+	if !allowedcmd.IsNixOS() {
+		return nil
+	}
+
+	interpreter, err := getInterpreter(executableLocation)
+	if err != nil {
+		return fmt.Errorf("getting interpreter for %s: %w", executableLocation, err)
+	}
+	interpreterLocation, err := findInterpreterInNixStore(interpreter)
+	if err != nil {
+		return fmt.Errorf("finding interpreter %s in nix store: %w", interpreter, err)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	cmd, err := allowedcmd.Patchelf(ctx, "--set-interpreter", interpreterLocation, executableLocation)
+	if err != nil {
+		return fmt.Errorf("creating patchelf command: %w", err)
+	}
+
+	if out, err := cmd.CombinedOutput(); err != nil {
+		return fmt.Errorf("running patchelf: output `%s`, error `%w`", string(out), err)
+	}
+
+	return nil
+}
+
+func getInterpreter(executableLocation string) (string, error) {
+	f, err := elf.Open(executableLocation)
+	if err != nil {
+		return "", fmt.Errorf("opening ELF file: %w", err)
+	}
+	defer f.Close()
+
+	interpSection := f.Section(".interp")
+	if interpSection == nil {
+		return "", errors.New("no .interp section")
+	}
+
+	interpData, err := interpSection.Data()
+	if err != nil {
+		return "", fmt.Errorf("reading .interp section: %w", err)
+	}
+
+	trimmedInterpData := bytes.TrimRight(interpData, "\x00")
+
+	// interpData should look something like "/lib64/ld-linux-x86-64.so.2" -- grab just the filename
+	return filepath.Base(string(trimmedInterpData)), nil
+}
+
+func findInterpreterInNixStore(interpreter string) (string, error) {
+	storeLocationPattern := filepath.Join("/nix/store/*glibc*/lib", interpreter)
```

Comment:
Hrm. I'm not sure this is correct.

I _think _ this is grabbing whatever glibc we find, but if that glibc was installed for some other package it might be removed. Is there a way to get the glibc that launcher has a dependency on?

Reply:
Hmm, okay -- I think we could run `patchelf --print-interpreter <path to current executable>` and assume that whatever `autoPatchelfHook` originally used is appropriate.

Reply:
I might be missing something... Does launcher declare a dependancy on glibc?

Reply:
Talking to Becca... I think this is still wrong, but I think solving it corretly might need more than we know today. We probably need to figure out how to declare runtime dependancies. To that end, I'm good merging and iterating